web/src/components/SaveMessage.js

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import moment from "moment";
+
+// Configure moment to display '1 time-unit ago' instead of 'a time-unit ago'
+// https://github.com/moment/moment/issues/3764
+moment.updateLocale("en", {
+  relativeTime: {
+    s: "seconds",
+    m: "1 minute",
+    mm: "%d minutes",
+    h: "1 hour",
+    hh: "%d hours",
+    d: "1 day",
+    dd: "%d days",
+    M: "1 month",
+    MM: "%d months",
+    y: "1 year",
+    yy: "%d years",
+  },
+});
+
+class SaveMessage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentMoment: moment(),
+    };
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(() => this.updateClock(), 1000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timerID);
+  }
+
+  updateClock() {
+    this.setState({
+      currentMoment: moment(),
+    });
+  }
+
+  render() {
+    const { lastSaved } = this.props;
+    const { currentMoment } = this.state;
+
+    const lastSavedMoment = moment(lastSaved);
+    const minutesOld = currentMoment.diff(lastSavedMoment, "minutes");
+    const minutesPerDay = 60 * 24;
+    const minutesPerYear = minutesPerDay * 365.25;
```

Comment:
Have you looked at [Moment's duration class](https://momentjs.com/docs/#/durations/)? I'm curious if it's `humanize()` method might accomplish this without us having to worry about leap years or all the other quirks of time.

Reply:
I'll take a look!